src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -297,7 +297,7 @@ private static RuntimeModule GetDynamicMethodsModule()
 
             // check and store the return value
             m_returnType = (returnType == null) ? (RuntimeType)typeof(void) : returnType.UnderlyingSystemType as RuntimeType;
-            if ((m_returnType == null) || !(m_returnType is RuntimeType) || m_returnType.IsByRef)
+            if ((m_returnType == null) || !(m_returnType is RuntimeType))
```

Comment:
You can delete `!(m_returnType is RuntimeType)` part of the condition as well while you are on it - it is always false because of `m_returnType` is RuntimeType.

Reply:
sure

Reply:
(And same a few lines above where the signature is checked.)